scrabble.js

```diff
@@ -1,8 +1,173 @@
-var Scrabble = function() {};
+var Scrabble = function() {
+    // due to how the specs are written, this score table does not account for the 50 point bonus for 7-letter words...
+    this.scoreTable = [
+        [1, "A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+        [2, "D", "G"],
+        [3, "B", "C", "M", "P"],
+        [4, "F", "H", "V", "W", "Y"],
+        [5, "K"],
+        [8, "J", "X"],
+        [10, "Q", "Z"]
+    ];
+};
+
+// score(word): returns the total score value for the given word. The word is input as a string (case insensitive).
+Scrabble.prototype.score = function(word) {
+    var wordScore = 0;
+    var nonAlphaChar = 0;
+    var alphaChars = ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T", "D", "G", "B", "C", "M", "P", "F", "H", "V", "W", "Y", "K", "J", "X", "Q", "Z"];
+
+    var scoreTable = this.scoreTable; //this local variable is necessary to bypass the error that otherwise would be thrown in line 18 because the scope of 'this' changes inside anonymous functions.
```

Comment:
The most common way to get beyond this issue is to create a variable called `self` which is set equal to `this`. Then, later on when you expect to use `this` you can use `self` instead. Sometimes folks also use `that` or `_this`, but I personally find both of those sort of confusing.